cilium-cli/connectivity/tests/pod.go

```diff
@@ -253,3 +256,167 @@ func (s *podToPodNoFrag) Run(ctx context.Context, t *check.Test) {
 
 	})
 }
+
+func PodToPodMissingIPCache(opts ...Option) check.Scenario {
+	options := &labelsOption{}
+	for _, opt := range opts {
+		opt(options)
+	}
+	return &podToPodMissingIPCache{
+		sourceLabels:      options.sourceLabels,
+		destinationLabels: options.destinationLabels,
+		method:            options.method,
+	}
+}
+
+type podToPodMissingIPCache struct {
+	sourceLabels      map[string]string
+	destinationLabels map[string]string
+	method            string
+}
+
+func (s *podToPodMissingIPCache) Name() string {
+	return "pod-to-pod-missing-ipcache"
+}
+
+type bpfMap struct {
+	ID int `json:"id"`
+}
+
+type bpfMapLookup struct {
+	Key   []string `json:"key"`
+	Value []string `json:"value"`
+}
+
+func (s *podToPodMissingIPCache) Run(ctx context.Context, t *check.Test) {
+	var i int
+	ct := t.Context()
+
+	var ciliumPod check.Pod
+	for _, pod := range ct.CiliumPods() {
+		ciliumPod = pod
+		break
+	}
+
+	// Collect all cilium_ipcache maps
+	ipcaches := []bpfMap{}
+	listIPCacheCmd := []string{"bpftool", "--json", "map", "list", "name", "cilium_ipcache"}
+	output, err := ciliumPod.K8sClient.ExecInPod(ctx, ciliumPod.Pod.Namespace, ciliumPod.Pod.Name, defaults.AgentContainerName, listIPCacheCmd)
+	if err != nil {
+		ct.Fatalf(`failed to execute "%s" in %s: %v`, listIPCacheCmd, ciliumPod.Pod.Name, err)
+	}
+	if err = json.Unmarshal(output.Bytes(), &ipcaches); err != nil {
+		ct.Fatalf(`failed to unmarshal output "%s": %v`, output, err)
+	}
+
+	// Delete echo pods entries from ipcache
+	for _, echo := range ct.EchoPods() {
+		echoIP := net.ParseIP(echo.Address(features.IPFamilyV4))
+		for _, ipcache := range ipcaches {
+			ipcacheKey := strings.Split(fmt.Sprintf("0x40 0x00 0x00 0x00 0x00 0x00 0x00 0x01 0x%02x 0x%02x 0x%02x 0x%02x 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00", echoIP[12], echoIP[13], echoIP[14], echoIP[15]), " ")
```

Comment:
I think this line could use a source code comment explaining a bit what the structure is. In particular

- What are the values around the IP address? I assume prefix length and padding for IPv6 but I honestly don' tknow
- Why are we using `echoIP[12:16]`. Does net.ParseIP return a 16 byte slice? Why? Could we use `echoIP.To4()` to compact it?